CONTRIBUTING.md

```diff
@@ -1,7 +1,175 @@
+
+### Step 2: Creating a branch for your changes
+
+It is a standard practice in git to create a new 'branch' (off 'master') for each new feature or bug fix. We will be creating a branch called 'my_feature' off of our local clone of the forked `jwst` repository:
+
+	1. Get your local 'master' branch up to date with any new changes upstream.
+		First, verify that you are currently on the `master` branch:
+			>> git status
+		and if your current branch isn't master,
+			>> git checkout master
+		Next, it is good practice to keep your local 'master' up-to-date with 'upstream/master'. To do this:
+			>> git pull upstream master
+		Note that `git pull` is sufficient here assuming you never make commits directly to 'master' - otherwise, you may have conflicts and this can get tricky. For this reason, it is recommended that you don't commit directly to master but rather branch off of it.
+
+	2. Now, create a new branch for your feature off your now up-to-date 'master' branch for your feature. You can call this branch whatever you like - in this example, we'll call it 'my_feature'
+
+		>> git checkout -b my_feature upstream/master
+
+		Doing this will create a new branch off local/master called 'my_feature', and automatically switch you over to your new branch.
+
+### Step 3: Installing `jwst` for development
+
+Next, you will create a new Python environment where you will install your new branch of `jwst`. We will show here how to use conda to manage environments, but there are other options.
+
+		1.  Create a conda environment.
+		It is good practice to maintain different environments for different versions of JWST and its dependencies. You will likely want to maintain one, for example, for the latest released version of JWST (i.e. what you get by doing `pip install jwst`), as well as one for development. Assuming the user has conda [installed](https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html), here we will create a new conda environment called 'jwst_dev' where we can install the new branch of our cloned repository.
+
+			>> conda create -n jwst_dev python
+
+		Doing this will create a new environment with just some basic packages (i.e setuptools, pip) installed.
+
+		2. Installing `jwst` in this environment
+		Make sure you are in your new environment:
+			>> conda activate jwst_dev
+		And now in the top level of your local `jwst` repository, ensuring you're on the 'my_feature' branch:
+			>> pip install -e .
+		This will install jwst from this cloned source code in 'editable' mode, meaning that you can import the code from this directory when within a python session. This makes it easier for development because you can have the code you're editing somewhere convenient in your file system vs. with other packages in 'site-packages'. If you cloned the repository on your Desktop, for example, you can modify it there and Python will know that is where the source code is when you're importing it within a Python session.
+
```

Comment:
Make a note it's a good idea to pip install "ipython" if they want to use it. Otherwise if ipython is run, it will pick up packages from the base env.